### PR TITLE
Implement get metric history bulk API in `mlflow server` for UI purposes

### PR DESCRIPTION
**Pull request by dbczumar**
## What changes are proposed in this pull request?

Implement get metric history bulk API in `mlflow server` for UI purposes

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [X] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/server/handlers.py [hunk 1]
```diff
@@ -1000,6 +1000,63 @@ def _get_metric_history():
...
+            message="GetMetricHistoryBulk request must specify a metric_key.",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+    metric_key = metric_key[0]
+
+    max_results = int(request_params.get("max_results", [MAX_HISTORY_RESULTS])[0])
+    max_results = min(max_results, MAX_HISTORY_RESULTS)
+
+    store = _get_tracking_store()
+    metrics_with_runids = []
```
Comment (apurva-koti) [inline mlflow/server/handlers.py:1036]:
```suggestion
    metrics_with_run_ids = []
```

File: mlflow/server/handlers.py [hunk 2]
```diff
@@ -1000,6 +1000,63 @@ def _get_metric_history():
...
+    for run_id in sorted(run_ids):
```
Comment (apurva-koti) [inline mlflow/server/handlers.py:1037]: Will this sort the list on every iteration? If so, can we sort it once into a var and loop over that?

**Review by apurva-koti — APPROVED** [review]
lgtm! some non-blocking comments

File: mlflow/server/handlers.py (hunk 2 above)
Comment (prithvikannan) [reply mlflow/server/handlers.py:1037]: I think this will only sort once

Comment (dbczumar) [reply mlflow/server/handlers.py:1037]: Only sorts once :)

File: mlflow/server/handlers.py [hunk 3]
```diff
@@ -1000,6 +1000,63 @@ def _get_metric_history():
...
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+
+    metric_key = request_params.get("metric_key", [])
+    if not metric_key:
+        raise MlflowException(
+            message="GetMetricHistoryBulk request must specify a metric_key.",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+    metric_key = metric_key[0]
```
[the remark this answers is not in the document]
Comment (dbczumar) [reply mlflow/server/handlers.py:1030]: Good question. They are *parsed* as such due to 

```
request_params = request.args.to_dict(flat=False)
```

(https://github.com/mlflow/mlflow/pull/7593/files#diff-5c10a4e2ca47745f06fa9e7201087acfc102849756cb8d85e774a5ac468cb037R1008).

The REST API contract uses scalars where appropriate, e.g. `max_results` is scalar.

[the remark this answers is not in the document]
Comment (dbczumar) [reply mlflow/server/handlers.py]: I've pushed an update to make argument parsing cleaner :)

File: mlflow/server/handlers.py (hunk 1 above)
Comment (dbczumar) [reply mlflow/server/handlers.py]: Applied this suggestion! Thanks @apurva-koti !

File: mlflow/server/handlers.py (hunk 2 above)
Comment (sunishsheth2009) [reply mlflow/server/handlers.py:1035]: Is there a technical reason we need to sort it and client need to map the run_ids then? Is there a reason we can't rely on the ordering?

**Review by sunishsheth2009 — APPROVED** [review]

File: mlflow/server/handlers.py (hunk 2 above)
Comment (dbczumar) [reply mlflow/server/handlers.py:1035]: The API contract (proto definition merged in Databricks) guarantees lexicographically ascending sort order for the returned metrics, so the backend implementation here needs to do the sorting.

**Pull request merged**